### ISSUE TITLE
redundantClosure leads to compile error

### ISSUE THREAD

**Issue by alistra** [nicklockwood/SwiftFormat#1236]
`redundantClosure` rule leads to a compile error.

This:
```
    static let once: Void = { object_setClass(Bundle.main, type(of: BundleExtension())) }()
```

Gets translated into this:
```
    static let once: Void = object_setClass(Bundle.main, type(of: BundleExtension()))
```

Leading to a type clash of `AnyClass?` not being `Void`

**Comment by nicklockwood** [comment]
@alistra this is a known issue that affects methods with `@discardableResult` return values. The workaround is to change the code to:

```swift
static let once: Void = { _ = object_setClass(Bundle.main, type(of: BundleExtension())) }()
```

or add a swiftformat comment directive:

```swift
// swiftformat:disable:next redundantClosure
static let once: Void = { object_setClass(Bundle.main, type(of: BundleExtension())) }()
```

**Comment by nicklockwood** [comment]
@alistra apologies, this is not actually mentioned in the list of known issues. I was confusing it with the `redundantVoidReturnType` rule. I'll add a note.

**Comment by nicklockwood** [comment]
@calda FYI.

**Comment by alistra** [comment]
I managed to do a workaround by disabling the rule. But I assume fix is planned soon? Or maybe this rule should be opt-in if it has potential to break stuff?

**Comment by calda** [comment]
Agree with your assessment that this is caused by the method being `@discardableResult`. One potential fix could be to not apply this rule in cases where the property is manually annotated with a `: Void` type.

**Comment by alistra** [comment]
@calda 

Probably also `Never` could be problematic?

**Comment by calda** [comment]
Yeah, we have to special-case methods that return `Never`: https://github.com/nicklockwood/SwiftFormat/pull/1098

**Comment by calda** [comment]
Here's a fix: https://github.com/nicklockwood/SwiftFormat/pull/1238

**Comment by nicklockwood** [comment]
@alistra fix landed in 0.49.14
